pkg/services/ngalert/notifier/channels/alertmanager.go

```diff
@@ -0,0 +1,106 @@
+package channels
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/alerting"
+	old_notifiers "github.com/grafana/grafana/pkg/services/alerting/notifiers"
+	"github.com/prometheus/alertmanager/template"
+	"github.com/prometheus/alertmanager/types"
+)
+
+// NewAlertmanagerNotifier returns a new Alertmanager notifier.
+func NewAlertmanagerNotifier(model *NotificationChannelConfig, t *template.Template) (*AlertmanagerNotifier, error) {
+	if model.Settings == nil {
+		return nil, alerting.ValidationError{Reason: "No settings supplied"}
+	}
+
+	urlStr := model.Settings.Get("url").MustString()
+	if urlStr == "" {
+		return nil, alerting.ValidationError{Reason: "Could not find url property in settings"}
+	}
+
+	var urls []*url.URL
+	for _, uS := range strings.Split(urlStr, ",") {
+		uS = strings.TrimSpace(uS)
+		if uS == "" {
+			continue
+		}
+
+		uS = strings.TrimSuffix(uS, "/") + "/api/v1/alerts"
+		u, err := url.Parse(uS)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse URL %q: %w", uS, err)
+		}
+
+		urls = append(urls, u)
+	}
+	basicAuthUser := model.Settings.Get("basicAuthUser").MustString()
+	basicAuthPassword := model.DecryptedValue("basicAuthPassword", model.Settings.Get("basicAuthPassword").MustString())
+
+	return &AlertmanagerNotifier{
+		NotifierBase: old_notifiers.NewNotifierBase(&models.AlertNotification{
+			Uid:                   model.UID,
+			Name:                  model.Name,
+			DisableResolveMessage: model.DisableResolveMessage,
+			Settings:              model.Settings,
+		}),
+		urls:              urls,
+		basicAuthUser:     basicAuthUser,
+		basicAuthPassword: basicAuthPassword,
+		message:           model.Settings.Get("message").MustString(`{{ template "default.message" .}}`),
+		logger:            log.New("alerting.notifier.prometheus-alertmanager"),
+		tmpl:              t,
+	}, nil
+}
+
+// AlertmanagerNotifier sends alert notifications to the alert manager
+type AlertmanagerNotifier struct {
+	old_notifiers.NotifierBase
+
+	urls              []*url.URL
+	basicAuthUser     string
+	basicAuthPassword string
+	message           string
+	logger            log.Logger
+	tmpl              *template.Template
+}
+
+// Notify sends alert notifications to Alertmanager.
+func (n *AlertmanagerNotifier) Notify(ctx context.Context, as ...*types.Alert) (bool, error) {
+	n.logger.Info("Sending Alertmanager alert", "alertmanager", n.Name)
+	if len(as) == 0 {
+		return true, nil
+	}
+
+	body, err := json.Marshal(as)
+	if err != nil {
+		return false, err
+	}
+
+	errCnt := 0
+	for _, u := range n.urls {
+		if _, err := sendHTTPRequest(ctx, u, httpCfg{
```

Comment:
What is the reason to not use `bus.DispatchCtx` here?

Reply:
@codesome the bus is being phased out. One huge issue with it is that it couples us to global state, which we're also getting rid of.

Reply:
Sounds like we need to modify all the other channels soon :)

Reply:
@codesome it's an incremental process :) Personally I mainly try to avoid adding more dependencies on it.